soc/riscv/riscv-ite/common/soc_irq.S

```diff
@@ -20,17 +20,11 @@ GTEXT(__soc_handle_irq)
 /*
  * SOC-specific function to handle pending IRQ number generating the interrupt.
  * Exception number is given as parameter via register a0.
+ * Jump to get_irq() function directly and return to caller by its
+ * ret instruction.
  */
 SECTION_FUNC(exception.other, __soc_handle_irq)
-	/* store ra */
-	addi sp, sp, -4
-	sw ra, 0(sp)
-	/* get interrupt number and clear interrupt status of soc */
-	call get_irq
-	/* restore ra */
-	lw ra, 0(sp)
-	addi sp, sp, 4
-	ret
+	j get_irq
```

Comment:
Is this change really necessary? How much space does it save? We shouldn't really use `j` for function calls here.

Reply:
I wonder if there's any chance that `get_irq` ends up too far to be called via `j` - in which case you'd need `jr` (or abuse `tail`).

Reply:
It should be safe to use `j` here, because the maximum flash size of it8xxx2 series is 1MB.

Reply:
Yes, `j` is indeed safe to jump within the 1MB range (section 2.5. of the [RISC-V spec](https://github.com/riscv/riscv-isa-manual/releases/download/draft-20220604-4a01cbb/riscv-spec.pdf)). Thanks answering my question.